src/iterative_ensemble_smoother/ies.cpp

```diff
@@ -287,7 +274,7 @@ create_transition_matrix(py::EigenDRef<MatrixXd> Y, py::EigenDRef<MatrixXd> R,
   if (ies_inversion == Inversion::exact) {
     exact_inversion(W, S, H, ies_steplength);
   } else {
-    subspace_inversion(W, ies_inversion, E, R, S, H, truncation,
+    subspace_inversion(W, ies_inversion, E, R.value(), S, H, truncation,
```

Comment:
Won`t `R.value()` raise a `[std::bad_optional_access](https://en.cppreference.com/w/cpp/utility/optional/bad_optional_access)` if R is not an actual matrix?

Reply:
Good question, thanks.
I've now made `R` optional in `subspace_inversion` and made some changes to how `R` is created in the first place.
I've also added tests to make sure every inversion method works when `observation_errors` are either a 1D array or a 2D array.